### PR TITLE
fix: location script simplified

### PR DESCRIPTION
**Pull request by ChrisSidebotham**
## Description

Replacement PR off new branch for #829 

This Pull Request is to introduce the region selector script used to rotate regional deployment as per the discussions within the core team. @AlexanderSehr FYI

[avm.res.web.static-site](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.web.static-site.yml)

### CODE REVIEW

**Comment by ChrisSidebotham** [comment]
Pending #838

File: avm/regions.json
Comment (eriqua) [inline avm/regions.json:1]: Is this file to be removed?

File: avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzAvailableResourceLocation.ps1 [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      "westus2",
+      "southcentralus",
+      "brazilsouth",
+      "westeurope",
+      "switzerlandnorth",
+      "asiasoutheast",
+      "japaneast",
+      "qatercentral",
+      "uaenorth",
+      "koreacentral"
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzAvailableResourceLocation.ps1:45]:
```suggestion
      "asiasoutheast",
      "brazilsouth",
      "eastus2",
      "japaneast",
      "koreacentral",
      "qatercentral",
      "southcentralus",
      "switzerlandnorth",
      "uaenorth",
      "westeurope",
      "westus2",
      "westus2"
```

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzAvailableResourceLocation.ps1:45]: Just sorting them...

File: avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzAvailableResourceLocation.ps1 [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  $formattedResourceProvider = ($formattedResourceType -split '[\/|\\]{1}')[0]
+  Write-Verbose "Resource type: $formattedResourceProvider"
+  $formattedServiceName = ($formattedResourceType -split '[\/|\\]{1}')[1]
+  Write-Verbose "Resource: $formattedServiceName"
+
+  $resourceRegionList = @()
+  $ResourceRegionList = (Get-AzResourceProvider | Where-Object { $_.ProviderNamespace -eq $formattedResourceProvider }).ResourceTypes | Where-Object { $_.ResourceTypeName -eq $formattedServiceName } | Select-Object -ExpandProperty Locations
+  Write-Verbose "Region list: $($resourceRegionList | ConvertTo-Json)"
+
+  $locations = Get-AzLocation | Where-Object { $_.DisplayName -in $ResourceRegionList } | Where-Object { $_.Location -notin $ExcludedRegions } | Where-Object { $_.PairedRegion -ne "{}" } | Where-Object { $_.RegionCategory -eq "Recommended" } |  Select-Object -ExpandProperty Location
```
Comment (AlexanderSehr) [inline avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzAvailableResourceLocation.ps1:68]:
```suggestion
  $locations = Get-AzLocation | Where-Object { 
    $_.DisplayName -in $ResourceRegionList  -and
    $_.Location -notin $ExcludedRegions -and
    $_.PairedRegion -ne "{}" -and
    $_.RegionCategory -eq "Recommended" 
  } |  Select-Object -ExpandProperty Location
```

File: avm/regions.json
Comment (ChrisSidebotham) [reply avm/regions.json:1]: Removed

**Comment by ChrisSidebotham** [comment]
deployment test running :)

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**